### PR TITLE
gst_all_1: use newScope to create package set

### PR DESCRIPTION
**Pull request by B4dM4n**
###### Motivation for this change

Overriding `gst_all_1.gstreamer` is a pain, because it is contained in a `rec` set and manually passes itself and other dependencies from the set directly to dependent packages. So in order to propagate a overwritten `gstreamer` to it's dependent packages an overlay like this is required:
```nix
final: prev: {
  gst_all_1 = prev.gst_all_1 // rec {
    gstreamer = prev.gst_all_1.gstreamer.overrideAttrs (old: { ... });

    gst-plugins-base = prev.gst_all_1.gst-plugins-base.override { inherit gstreamer; };

    gst-plugins-good = prev.gst_all_1.gst-plugins-good.override { inherit gst-plugins-base; };

    ...
  };
}
```
This overlay would also miss new packages in the set depending on any of the current packages.

Using `lib.makeScope` would allow to conveniently override any package in the set and ensure it is used in all dependent packages:

```nix
final: prev: {
  gst_all_1 = prev.gst_all_1.overrideScope' (final: prev: {
    gstreamer = prev.gstreamer.overrideAttrs (old: { ... });
  });
}
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2534,9 +2534,7 @@ in
 
   monetdb = callPackage ../servers/sql/monetdb { };
 
-  monado = callPackage ../applications/graphics/monado {
-    inherit (gst_all_1) gstreamer gst-plugins-base;
-  };
+  monado = gst_all_1.callPackage ../applications/graphics/monado {  };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix:2537]: I think we should avoid this pattern in package sets: using the `callPackage` from a different scope breaks composition. In these cases it is in my opinion much better to pass in the subpackage set.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by jtojnar** [comment]
Alternately, we could move those packages to top-level. But scoping sound good to me too. cc @worldofpeace

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (FRidh) [reply pkgs/top-level/all-packages.nix:2537]: Related RFC https://github.com/NixOS/rfcs/pull/83.

**Comment by worldofpeace** [comment]
@jtojnar I think some time in the past came upon that the need for a `gst_all_1` has passed. It was really confusing to have old versions of gst as the top level attributes. I would be in favor of moving them all to top-level and keeping `gst_all_1` as an alias.

**Comment by Princemachiavelli** [comment]
I think #131295 fixed this by removing `rec`. I'm able to easily override `gstreamer` with the following overlay (there might be an even better way but this makes sense to me):
```nix
final: prev: {
  gst_all_1 = prev.gst_all_1 // {
      gstreamer = prev.gst_all_1.gstreamer.overrideAttrs( prev: {
        pname= prev.pname + "-foo";
      });
  };
}
```
Checking `gst_all_1.gst-plugins-base` shows that the override to `gstreamer` is pulled in without having to manual pass the new `gstreamer.

**Pull request closed without merging**